egs/wsj/s5/utils/combine_data.sh

```diff
@@ -39,6 +39,21 @@ for dir in $*; do
   fi
 done
 
+# W.r.t. utt2uniq file the script has different behavior compared to other files
+# it is not compulsary for it to exist in all directories
+# when it does not exist we have to create it
+# check if utt2uniq is in the extra_files
+for file in $extra_files; do
+  if [ "$file" == "utt2uniq" ]; then
+    for in_dir in $*; do
+      if [ ! -f $in_dir/utt2uniq ]; then
+        # we assume that utt2uniq is a one to one mapping
+        cat $in_dir/utt2spk | awk '{printf("%s %s\n", $1, $1);}' > $in_dir/utt2uniq
+      fi
```

Comment:
I don't think creating the utt2uniq files in the source directories is the right solution-- you should have the conditional at the time you combine them-- if you do combine them.
Conceptually, the source directories are 'const'.  bash doesn't have 'const', but in our heads they are const.